sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusProcessorClient.java

```diff
@@ -224,9 +223,8 @@ public synchronized void start() {
         receiveMessages();
 
         // Start an executor to periodically check if the client's connection is active
-        if (this.scheduledExecutor == null) {
-            this.scheduledExecutor = Executors.newSingleThreadScheduledExecutor();
-            scheduledExecutor.scheduleWithFixedDelay(() -> {
+        if (monitorDisposable == null) {
+            monitorDisposable = Schedulers.boundedElastic().schedulePeriodically(() -> {
```

Comment:
Can we write a small note on why we choose this over `Flux.interval` that -  _the restart route involves tearing down multiple levels of clients synchronously. The boundedElastic is used instead of the parallel (single-threaded scheduler backing `Flux.interval`), so that we don't block the parallel threads for so long_

\\cc @conniey any thoughts/concern?

Reply:
I agree, the explanation is definitely needed here. I'll add the note for not choosing Flux.interval.